### ISSUE TITLE
When the last_round_number of the validator is inconsistent, grandpa cannot continue the consensus.

### ISSUE THREAD

**Issue by gguoss** [paritytech/substrate#1793]
# How to reproduce:
Run substrate in local chain spec.
when Alice  and Bob validator's last_round_number > 0 ,  then delete Bob validator's db, then restart Bob validator. 
Alice and Bob's last_round_number are inconsistent,  so grandpa can't finalize.

**Comment by rphmeier** [comment]
I would classify this as expected behavior.

**Comment by gguoss** [comment]
https://github.com/paritytech/substrate/blob/6a4b58068421f280ff2dfe564220c58147c35f4a/core/finality-grandpa/src/lib.rs#L555

When there are multiple validator, the initial status of the grandpa is also inconsistent, and the validator cannot communicate vot.

**Comment by gguoss** [comment]
But there is still a small problem：
When there are problems with more than 1/3 of the validator nodes, grandpa cannot be recovered. Delete the data directly, then sync, and it will not work. Unless you go directly to copy the same correct db.

**Comment by gguoss** [comment]
In other words, when synchronizing, it is impossible to synchronize the state of the latest grandpa, only the finalize count can be synchronized.

**Comment by gguoss** [comment]
This will cause the nodes in the last_state and last_state -1 states to have no consensus.

**Comment by gguoss** [comment]
Finalize information is passively broadcast and cannot be actively obtained from connected nodes.

This will cause all new sync nodes to not synchronize any finalize information when grandpa is not working properly.

**Comment by gavofyork** [comment]
Agree. Obviously it should be possible for nodes to recover. Its reasonable for grandpa to halt as long as Bob is offline, but saying that it can never recover once the database is emptied seems needlessly defeatist.

**Comment by rphmeier** [comment]
I'm not convinced it's possible without keeping all messages forever.

**Comment by gguoss** [comment]
if 4 validators : (A, B , C, D )
A:  last_state
B:  last_state - 1
C: last_state - 2
D: last_state - 2

Can this continue the consensus?
If not, in this case, obviously this is very likely to happen.

**Comment by rphmeier** [comment]
yes, it can, as long as only one validator is offline at a time (offline including time spent catching up to consensus)

**Comment by gguoss** [comment]
if 4 validator: (A, B, C, D)
A, B , C  run for a while, 
last_state(A) = last_state(B) = last_state(C) > 0 , last_state(D) = 0
Force C to go offline， Start running D.

Result:
Grandpa can't finalize.

Run C again
Result:
Grandpa can't finalize.

Only restart running(A or B or C or D), grandpa can finalize.

**Comment by rphmeier** [comment]
Yes, this is expected behavior, as I have described. A 1/3 BFT system with 4 participants cannot tolerate more than one fault at a time. I am not sure what you expect us to do, because this is a theoretical limit.

**Comment by rphmeier** [comment]
https://github.com/paritytech/finality-grandpa/issues/47

**Comment by andresilva** [comment]
This shouldn't happen anymore as Bob would send a catch-up request to Alice and then proceed to the latest round.
